diff --git a/src/modules/HMRClient.js b/src/modules/HMRClient.js
--- a/src/modules/HMRClient.js
+++ b/src/modules/HMRClient.js
@@ -3,6 +3,7 @@
 const EventEmitter = require("./vendor/eventemitter3");
 const HEARTBEAT_INTERVAL_MS = 20_000;
 const inject = ({ module: [id, code], sourceURL }) => {
+  global.__workletsModuleProxy?.propagateModuleUpdate?.(code, sourceURL);
   if (global.globalEvalWithSourceUrl) {
     global.globalEvalWithSourceUrl(code, sourceURL);
   } else {
